lib/naughty/core.lua

```diff
@@ -239,6 +239,51 @@ function naughty.suspend()
     suspended = true
 end
 
+local conns = {}
+
+--- Connect a global signal on the notification engine.
+--
+-- Functions connected to this signal source will be executed when any
+-- notification object emits the signal.
+--
+-- It is also used for some generic notification signals such as
```

Comment:
Don't we have a common implementation of `connect_signal` and friends in `gears.object`? Why do we need another implementation of that instead of just using that one?

Reply:
In this case because it is a module and not a class. However, if you prefer, I can move the class connect signal implementation to `gears`. And yes, the only difference is `:` vs. `.`, but for consistency with the other modules, `.` is required.